lib/elixir/test/elixir/option_parser_test.exs

```diff
@@ -256,6 +256,12 @@ defmodule OptionParserTest do
            == {[foo: true, boo: "-"], ["-"], []}
   end
 
+  test "allow nonexistent atom`s" do
+    opts = [switches: [a: :string], allow_nonexistent_atoms: true]
+    assert OptionParser.parse(["--a", "b", "--c", "d", "--option-key-creates-atom", "f"], opts)
+           == {[{:a, "b"}, {:c, "d"}, {String.to_atom("option_key_creates_atom"), "f"}], [], []}
```

Comment:
Let's focus on testing `:allow_nonexistent_atoms` functionality: we can simply pass `["--option-key-creates-atom"]` as input.